beaconChain/lib/powChain/index.ts

```diff
@@ -0,0 +1,96 @@
+import { ethers, Contract } from 'ethers';
+import { EtherscanProvider } from 'ethers/providers';
+import { Deposit, DepositData, DepositInput } from '../../interfaces/blocks';
+import { ChainStart } from "../../interfaces/state";
+import { MAINNET_DEPOSIT_ADDRESS, DEPOSIT_CONTRACT_ADDRESS, DEPOSIT_CONTRACT_BLOCK_NUMBER, DEPOSIT_CONTRACT_ABI } from "../../constants/constants";
+
+// Type stubs
+type uint64 = number;
+type hash32 = string;
+
+
+// NOTE: This is stubbed
+// TODO: We should find a way to reject the promise somehow.
+// TODO: There is porbably a better way to scrape the logs.
+/**
+ * Polls the eth1.x deposit contract for validator deposits. If the ChainStart log is emitted, return the promise with
+ * intial state data.
+ * @returns {Promise<ChainStart>}
+ */
+const waitForChainStart = () => {
+  return new Promise<ChainStart>((resolve) => {
+    const deposits: Deposit[] = [];
+
+    // Connect to the network
+    let provider = ethers.getDefaultProvider();
+
+    // Deposit Contract
+    let depositContract: ethers.Contract = new ethers.Contract(MAINNET_DEPOSIT_ADDRESS, DEPOSIT_CONTRACT_ABI, provider);
+
+    // Eth1Deposit log filter
+    let depositTopic: string = depositContract.Eth1Deposit();
+    let depositFilter: ethers.EventFilter = {
+      address: MAINNET_DEPOSIT_ADDRESS,
+      topics: [ depositTopic ]
+    };
+
+    // ChainStart log filter
+    let chainStartTopic: string = depositContract.ChainStart();
+    let chainStartFilter: ethers.EventFilter = {
+      address: MAINNET_DEPOSIT_ADDRESS,
+      topics: [ chainStartTopic ]
+    };
+
+    // Listen for Eth1Deposit logs
+    provider.on(depositFilter, (previousReceiptRoot: hash32[], data: DepositData, totalDepositcount: uint64, log) => {
+      const newDeposit: Deposit = formatDeposit(previousReceiptRoot, data, totalDepositcount);
+      deposits.push(newDeposit);
```

Comment:
Whats the point of storing the deposits here? Should we maybe provide a function to allow their retrieval instead of discarding them in this code block?

Reply:
Not sure what you mean by discarding...
This is where classes could be very useful, and we can store to a state variable. But the objective here is we need to retrieve the deposits until chainstart emits then we take the formatted deposit[] and pass it to the next function in `start` to generate initial state

Reply:
Correct me if I'm wrong, deposits will only added until the ChainStart event resolves the promise. ie. future deposits won't he captured here.

Reply:
Correct, that's why I created the deposits helper. Um unsure with the current spec at what point a new deposit needs to be captured. I.e what point in the state transition.